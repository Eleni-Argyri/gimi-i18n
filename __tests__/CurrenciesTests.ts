import {getSupportedCurrencyInfos, getDefaultCurrencyCode} from '../index'
import _DefaultCurrencies from '../DefaultCurrencies'
import _ExchangeRates from '../ExchangeRates'
import Decimal from 'decimal.js-light'

const DefaultCurrencies = _DefaultCurrencies as Record<string, any>
const ExchangeRates = _ExchangeRates as Record<string, any>

jest.disableAutomock()

describe('Currencies', () => {
  it('it should return supported supportedCurrencyCodes', () => {
    expect(getSupportedCurrencyInfos()).not.toBeUndefined()
    expect(getSupportedCurrencyInfos()).toHaveLength(12)
    expect(getSupportedCurrencyInfos()).toMatchSnapshot()
  })

  it('it should return default currency', () => {
    expect(getDefaultCurrencyCode('FI')).toEqual('EUR')
    expect(getDefaultCurrencyCode('NO')).toEqual('NOK')
    expect(getDefaultCurrencyCode('GB')).toEqual('GBP')
    expect(getDefaultCurrencyCode('SE')).toEqual('SEK')
    expect(getDefaultCurrencyCode('BE')).toEqual('EUR')
    expect(getDefaultCurrencyCode()).toEqual('EUR')
  })

  it('all currencies in Default CurrencyCodes needs to be in supported Currency Codes', () => {
    Object.values(DefaultCurrencies).forEach((currencyCode) =>
      expect(getSupportedCurrencyInfos().find((currencyInfo) => currencyInfo.code === currencyCode)).toBeDefined()
    )
  })

  Object.keys(DefaultCurrencies).forEach((key) => {
    it(`Should be an exchange rate for ${key} (entry found in default currencies)`, () => {
      expect(ExchangeRates[DefaultCurrencies[key]]).toBeDefined()
    })
  })

  Object.values(getSupportedCurrencyInfos()).forEach((val) => {
    it(`All defined currencies should have conversion rate ${
      ExchangeRates[val.code]
    } `, () => {
      expect(ExchangeRates[val.code]).toBeDefined()
    })

    it(`All defined currencies should be a positive number ${
      ExchangeRates[val.code]
    } `, () => {
      let exchangeRate = new Decimal(ExchangeRates[val.code])

      expect(exchangeRate.isPositive()).toBeTruthy()
      expect(ExchangeRates[val.code]).toBeDefined()
    })
  })
})
