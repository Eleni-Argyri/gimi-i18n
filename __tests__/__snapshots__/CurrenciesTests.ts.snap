// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies it should return supported supportedCurrencyCodes 1`] = `
[
  {
    "code": "SEK",
    "langKey": "currency_sek",
    "name": "Swedish Krona",
  },
  {
    "code": "NOK",
    "langKey": "currency_nok",
    "name": "Norwegian Krone",
  },
  {
    "code": "DKK",
    "langKey": "currency_dkk",
    "name": "Danish Krone",
  },
  {
    "code": "GBP",
    "langKey": "currency_gbp",
    "name": "Pound Sterling",
  },
  {
    "code": "USD",
    "langKey": "currency_usd",
    "name": "US Dollar",
  },
  {
    "code": "EUR",
    "langKey": "currency_eur",
    "name": "Euro",
  },
  {
    "code": "INR",
    "langKey": "currency_inr",
    "name": "Indian Rupee",
  },
  {
    "code": "AUD",
    "langKey": "currency_aud",
    "name": "Australian Dollar",
  },
  {
    "code": "CAD",
    "langKey": "currency_cad",
    "name": "Canadian Dollar",
  },
  {
    "code": "NZD",
    "langKey": "currency_nzd",
    "name": "New Zealand Dollar",
  },
  {
    "code": "ISK",
    "langKey": "currency_isk",
    "name": "Iceland Krona",
  },
  {
    "code": "CHF",
    "langKey": "currency_chf",
    "name": "Swiss Franc",
  },
]
`;
